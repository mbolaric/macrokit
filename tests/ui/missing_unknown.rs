use macrokit::FromReprWithUnknown;

#[derive(Debug, PartialEq, FromReprWithUnknown)]
#[repr(u8)]
pub enum Status {
    Active = 0,
    Inactive = 1,
    // Missing the 'Unknown' variant
}

fn main() {}
