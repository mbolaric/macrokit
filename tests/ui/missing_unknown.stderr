error[E0599]: no variant or associated item named `Unknown` found for enum `Status` in the current scope
 --> tests/ui/missing_unknown.rs:3:28
  |
3 | #[derive(Debug, PartialEq, FromReprWithUnknown)]
  |                            ^^^^^^^^^^^^^^^^^^^ variant or associated item not found in `Status`
4 | #[repr(u8)]
5 | pub enum Status {
  | --------------- variant or associated item `Unknown` not found for this enum
  |
  = note: this error originates in the derive macro `FromReprWithUnknown` (in Nightly builds, run with -Z macro-backtrace for more info)
